src/main/java/com/fasterxml/jackson/databind/util/JSONPObject.java

```diff
@@ -78,6 +88,7 @@ public void serialize(JsonGenerator jgen, SerializerProvider provider)
             provider.findTypedValueSerializer(cls, true, null).serialize(_value, jgen, provider);
         }
         jgen.writeRaw(')');
+        jgen.setCharacterEscapes(currentCharacterEscapes);
```

Comment:
I think this should go in `finally` block, just in case? Probably not big difference but as a matter of principle.